src/modules/Dropdown/Dropdown.js

```diff
@@ -686,10 +702,12 @@ export default class Dropdown extends Component {
       search && searchQuery && 'filtered'
     )
     let _text = placeholder
-    if (text && !searchQuery) {
-      _text = text
-    } else if (searchQuery) {
+    if (searchQuery) {
       _text = null
+    } else if (text) {
+      _text = text
+    } else if (open && !multiple) {
+      _text = this.getSelectedItemText()
```

Comment:
That is indeed better. I made a little change and used `_.get(options, `[${selectedIndex}]`)` instead of `options[selectedIndex]`, since the index could initially be `undefined`.